iOS/MSMapsModules/MSMapsModules/MSMapGeoJsonParser/MSMapGeoJsonParser.m

```diff
@@ -0,0 +1,186 @@
+//
+//  MSMapGeoJsonParser.m
+//  MSMapsModules
+//
+//  Created by Elizabeth Bartusiak (t-elbart) on 6/12/20.
+//  Copyright © 2020 Microsoft Corporation.
+//  Licensed under the MIT license.
+//
+
+#import "MSMapGeoJsonParser.h"
+#import <MicrosoftMaps/MicrosoftMaps.h>
+
+@implementation MSMapGeoJsonParser {
+  MSMapElementLayer *mLayer;
+};
+
++ (nullable MSMapElementLayer *)parse:(nonnull NSString *)geojson
+                                error:(NSError * _Nullable *)error {
+  if (geojson == nil || geojson.length == 0) {
+    if (error != nil) {
+      *error = [NSError
+          errorWithDomain:
+              @"com.microsoft.modules.MSMapsModules.InvalidArgumentError"
+                     code:-100
+                 userInfo:@{
+                   NSLocalizedDescriptionKey : @"Input String cannot be null."
+                 }];
+    }
+    return nil;
+  }
+
+  MSMapGeoJsonParser *instance = [[MSMapGeoJsonParser alloc] init];
+  NSError *localError;
+  MSMapElementLayer *layer = [instance internalParse:geojson error:&localError];
+  if (localError != nil) {
+    if (error != nil) {
+      *error = localError;
+    }
+    return nil;
+  }
+  return layer;
+}
+
+- (nullable MSMapElementLayer *)internalParse:(nonnull NSString *)geojson
+                                        error:(NSError * _Nullable *)error {
+  mLayer = [[MSMapElementLayer alloc] init];
+  NSData *jsonData = [geojson dataUsingEncoding:NSUTF8StringEncoding];
+  NSError *jsonError;
+  NSDictionary *jsonObject =
+      [NSJSONSerialization JSONObjectWithData:jsonData
+                                      options:kNilOptions
+                                        error:&jsonError];
+
+  if (jsonObject == nil) {
+    *error = [NSError
+        errorWithDomain:@"com.microsoft.modules.MSMapsModules.JSONError"
+                   code:-200
+               userInfo:@{NSLocalizedDescriptionKey : @"Malformed JSON."}];
+    return nil;
+  }
+
+  NSString *type =
+      [NSString stringWithFormat:@"%@", [jsonObject objectForKey:@"type"]];
+  if ([type isEqualToString:@"FeatureCollection"]) {
+    // TODO: parseFeatureCollection
+  } else {
+    if ([type isEqualToString:@"Feature"]) {
+      // TODO: feature
+    }
+    [self switchToType:jsonObject error:error];
+  }
+
+  return mLayer;
+}
+
+- (void)switchToType:(nonnull NSDictionary *)object
+               error:(NSError * _Nullable *)error {
+  NSString *type =
+      [NSString stringWithFormat:@"%@", [object objectForKey:@"type"]];
+  if ([type isEqualToString:@"Polygon"]) {
+    // TODO: parsePolygon
+  } else if ([type isEqualToString:@"Point"]) {
+    NSArray *coordinates = [self getCoordinates:object error:error];
+    if (coordinates != nil) {
+      [self parsePoint:coordinates error:error];
+    } else {
+      mLayer = nil;
```

Comment:
Remove this pattern. It's difficult to verify that all the code afterwards will work.